### PR TITLE
Feature/add share button

### PR DESCRIPTION
**Pull request by kaminskypavel**
added a share button to make it easier to share within the app.

the link sent is respecting the locale

image

image

### CODE REVIEW

File: src/components/common/ShareAppButton.tsx [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 import { Icon, TouchableOpacity } from '.';
-import { toggleChangeLanguage } from '../../actions/LocaleActions';
-
 
 const ShareAppButton: ElementType = () => {
+  const onShare = async () => {
+    try {
+      await Share.share({
+        message:
+                    'Hey, Please download. https://govextra.gov.il/ministry-of-health/corona/corona-virus',
```
Comment (ItayElgazar) [inline src/components/common/ShareAppButton.tsx]: Maybe it's better to also support localization here based on the selected language

Comment (kaminskypavel) [reply src/components/common/ShareAppButton.tsx]: the link the is already globalized. regarding the "Hey ,Please donwload" text this should be part of the i18n effort opened on other issue

Comment (kaminskypavel) [reply src/components/common/ShareAppButton.tsx]: PS - you're watching an outdated commit

Comment (ItayElgazar) [reply src/components/common/ShareAppButton.tsx]: Yes, I realized that. And still, regarding the "Hey, Please download" do we have some global ticket for i18n issues?

Comment (kaminskypavel) [reply src/components/common/ShareAppButton.tsx]: there are several. French (feat), Russian(fix) and Spanish. 
this is a feature branch, i strongly suggest we move all i18n discussions to another issue.

Comment (ItayElgazar) [reply src/components/common/ShareAppButton.tsx]: But this is jus features or fixes to add or fix some Grammarly typos in the language, it's not related to the fact that your string should be added to the English i18n, it's part of the feature

Comment (ItayElgazar) [reply src/components/common/ShareAppButton.tsx]: I mean, add the text to the English, Hebrew section here: https://github.com/MohGovIL/hamagen-react-native/blob/master/src/locale/texts.json

It's better than having hard-coded text

Comment (kaminskypavel) [reply src/components/common/ShareAppButton.tsx]: there are several issues with us changing the text here.

1. the text probably will be rephrased in each language differently, so we can use google translate, but it's preferred that the language maintainers would make sure it makes sense.
2. unless we know the recipient language, I think sticking with English is a reasonable idea.

Comment (kaminskypavel) [reply src/components/common/ShareAppButton.tsx]: I can add the texts, its no brainer, can someone from the team can confirm the phrasing? or supply translations?

Comment (ItayElgazar) [reply src/components/common/ShareAppButton.tsx]: 1. You're right, that's why I said it's better to at least have English and Hebrew and for the rest, we should tag other maintainers for help.

2. Probably the recipient language is the same language as the one that shares the app.
But...It's just my opinion.

Anyway, good job. this feature is necessary, especially for elders.

Comment (kaminskypavel) [reply src/components/common/ShareAppButton.tsx]: thanks man, appreciated 🙏

File: src/components/common/ShareAppButton.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  locale: 'he'|'en'|'ar'|'am'|'ru',
+}
+
+const ShareAppButton: ElementType = ({ locale }:Props) => {
+  const onShare = async () => {
+    try {
+      const downloadLink = config.downloadMohLandingPage[locale];
+
+      await Share.share({
+        message: `Hey, Please download. ${downloadLink}`,
```
Comment (RoySegall) [inline src/components/common/ShareAppButton.tsx]: What about translating the message?

File: src/config/config.ts [hunk 3]
```diff
@@ -46,6 +46,13 @@ const config: AppIds = {
       ru: 'https://govextra.gov.il/ministry-of-health/corona/corona-virus-ru',
       ar: 'https://govextra.gov.il/ministry-of-health/corona/corona-virus-ar/'
     },
+    downloadMohLandingPage: {
+      he: 'https://govextra.gov.il/ministry-of-health/hamagen-app/download-he/',
+      en: 'https://govextra.gov.il/ministry-of-health/hamagen-app/download-en/',
+      ru: 'https://govextra.gov.il/ministry-of-health/hamagen-app-ru/download-ru/',
+      ar: 'https://govextra.gov.il/ministry-of-health/hamagen-app/download-ar/',
+      am: 'https://govextra.gov.il/ministry-of-health/hamagen-app/download-he/',
```
Comment (aharonha) [inline src/config/config.ts]: fix the url

File: src/components/common/ShareAppButton.tsx (hunk 2 above)
Comment (kaminskypavel) [reply src/components/common/ShareAppButton.tsx]: please see my conversation with @ItayElgazar

File: src/config/config.ts [hunk 4]
```diff
@@ -96,6 +103,13 @@ const config: AppIds = {
...
```
Comment (aharonha) [inline src/config/config.ts]: fix the url (if exits such url)

**Review by aharonha — COMMENTED** [review]
In  `am` locale the URL is not right. Is the page missing?

File: src/config/config.ts (hunk 4 above)
Comment (kaminskypavel) [reply src/config/config.ts]: I couldn't find one. please supply a link

Comment (ofekashery) [reply src/config/config.ts]: @kaminskypavel https://govextra.gov.il/ministry-of-health/hamagen-app-am/download-am

Comment (kaminskypavel) [reply src/config/config.ts]: updated. thanks

File: src/config/config.ts [hunk 5]
```diff
@@ -146,6 +160,13 @@ const config: AppIds = {
       ru: 'https://govextra.gov.il/ministry-of-health/corona/corona-virus-ru',
       ar: 'https://govextra.gov.il/ministry-of-health/corona/corona-virus-ar'
     },
+    downloadMohLandingPage: {
+      he: 'https://govextra.gov.il/ministry-of-health/hamagen-app/download-he/',
+      en: 'https://govextra.gov.il/ministry-of-health/hamagen-app/download-en/',
+      ru: 'https://govextra.gov.il/ministry-of-health/hamagen-app-ru/download-ru/',
+      ar: 'https://govextra.gov.il/ministry-of-health/hamagen-app/download-ar/',
+      am: 'https://govextra.gov.il/ministry-of-health/hamagen-app/download-he/',
```
Comment (shaked113021) [inline src/config/config.ts]: Should be download-am

File: src/config/config.ts (hunk 4 above)
Comment (aharonha) [reply src/config/config.ts]: This block is redundant

Comment (shaked113021) [reply src/config/config.ts]: Please change this url to the url used above. This is the url for the Hebrew download page, url for Amharic download page is needed here.

File: src/types/index.ts [hunk 6]
```diff
@@ -24,6 +24,13 @@ export interface Config {
     ar: { title: string, body: string },
     duration: number
   },
+  downloadMohLandingPage: {
```
Comment (shaked113021) [inline src/types/index.ts]: I know it's a minor thing but please organize this block in the same way the other download blocks organized:
he
en
ru
ar
am

File: src/config/config.ts (hunk 5 above)
Comment (shaked113021) [reply src/config/config.ts]: Understood,But please change to Amharic download url please(it is currently a Hebrew one)

**Comment by kaminskypavel** [comment]
Thanks for the reviews. But I'd be happy to see any response from the maintainer first.

**Comment by ItayElgazar** [comment]
@MiriShalev This is an important feature

**Comment by MiriShalev** [comment]
@kaminskypavel We intend to add this feature at the next version

**Pull request closed without merging**